lms/djangoapps/edraak_university/tests/test_views.py

```diff
@@ -0,0 +1,284 @@
+
+
+class CourseContextMixinTest(ModuleStoreTestCase):
+    """
+    Tests for CourseContextMixin context_data and helper methods.
+    """
+
+    def setUp(self):
+        super(CourseContextMixinTest, self).setUp()
+
+        course = CourseFactory.create()
+        user = UserFactory.create()
+
+        # Mocks Django generic view
+        base_view_mock_class = type('BaseViewMock', (object,), {
+            'get_context_data': lambda _: {},
+        })
+
+        # Mocks a University ID view
+        view_mock_class = type('ViewMock', (CourseContextMixin, base_view_mock_class,), {
+            'kwargs': {
+                'course_id': unicode(course.id)
+            },
+            'request': Mock(
+                user=user,
+            ),
+        })
+
+        self.view = view_mock_class()
+        self.course = course
+        self.user = user
+
+    def test_get_course(self):
+        course_key = self.view.get_course_key()
+        self.assertIsInstance(course_key, CourseLocator)
+
+        data = self.view.get_context_data()
+        self.assertEquals(data['course'].id, self.course.id)
+
+    def test_access_control(self):
+        with self.assertRaises(Http404):
+            # By default the user is not 'staff'
+            self.view.require_staff_access()
+
+        try:
+            with patch('edraak_university.mixins.has_access', return_value=True):
+                self.view.require_staff_access()
+        except Http404:
+            self.fail('Should not raise a 404 error for staff users.')
+
+
+@ddt.ddt
+class UniversityIDViewStudentTest(ModuleStoreTestCaseLoggedIn):
+    """
+    Integration tests for the University ID student views.
+    """
+
+    ENROLL_USER = True
+    LOGIN_STAFF = False
+
+    def setUp(self):
+        super(UniversityIDViewStudentTest, self).setUp()
+        self.url = reverse('edraak_university_id', args=[unicode(self.course.id)])
+
+    def create_course(self):
+        """
+        Overrides `create_course` to enable university ID.
+        """
+        return CourseFactory.create(enable_university_id=True)
+
+    def submit_form(self, data_overrides=None):
+        """
+        Submits the student university ID form and returns the response.
+        """
+
+        form_data = {
+            'full_name': self.user.profile.name,
+            'university_id': 'A100C50',
+            'section_number': 'B',
+        }
+
+        if data_overrides:
+            form_data.update(data_overrides)
+
+        return self.client.post(self.url, data=form_data)
+
+    def test_basic_request(self):
+        res = self.client.get(self.url)
+        self.assertContains(res, 'university-id-form')
+        self.assertContains(res, 'class="important"', msg_prefix='No information has been posted yet!')
+
+        soup = BeautifulSoup(res.content)
+        self.assertFalse(soup.select('ul.errorlist'), msg='Initial form should not contain errors')
+
+    @patch('edraak_university.views.update_account_settings')
```

Comment:
This is a patch for the `update_account_settings` API. Due to the way  `mock.patch` works, it is always reliable to patch the local imported function (`edraak_university.views.update_account_settings`) instead of patching the original one (`user_api.accounts.api.update_account_settings`).

I'm not sure what do you mean by the `account_settings` field.